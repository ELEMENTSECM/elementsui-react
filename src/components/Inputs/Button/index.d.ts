import * as React from 'react';

export interface ButtonProps {
	/**
	 * Button label
	 */
	label?: string;
	/**
	 * Button is disabled
	 */
	disabled?: boolean;
	/**
	 * Mouse click event handler
	 */
	onCLick?: (...args: any[]) => any;
	/**
	 * Primary button
	 */
	isPrimary?: boolean;
}

export const Button: React.SFC<ButtonProps>;
