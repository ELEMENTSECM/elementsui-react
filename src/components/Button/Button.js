import * as React from 'react';
import { DefaultButton } from 'office-ui-fabric-react/lib/Button';
import PropTypes from 'prop-types';

/** Button example */
function Button({ label, disabled = false }) {
	return (
		<DefaultButton disabled={disabled} primary={true}>
			{label}
		</DefaultButton>
	);
}

Button.propTypes = {
	/** Button label */
	label: PropTypes.string,
	/** Button is disabled */
	disabled: PropTypes.bool
};

export default Button;
