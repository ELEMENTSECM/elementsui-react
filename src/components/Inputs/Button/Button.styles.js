export const getStyles = props => {
	const { className, theme } = props;
	const { palette, semanticColors } = theme;

	return {
		root: ['ms-df', {}, className]
	};
};
