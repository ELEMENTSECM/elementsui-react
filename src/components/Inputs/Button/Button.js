import * as React from 'react';
import PropTypes from 'prop-types';
import { DefaultButton } from 'office-ui-fabric-react/lib/Button';

/** Button example */
function Button({ label, disabled = false, onClick, isPrimary }) {
	return (
		<DefaultButton disabled={disabled} primary={isPrimary} onClick={onClick}>
			{label}
		</DefaultButton>
	);
}

Button.propTypes = {
	/** Button label */
	label: PropTypes.string,
	/** Button is disabled */
	disabled: PropTypes.bool,
	/** Mouse click event handler */
	onCLick: PropTypes.func,
	/** Primary button */
	isPrimary: PropTypes.bool
};

export default Button;
