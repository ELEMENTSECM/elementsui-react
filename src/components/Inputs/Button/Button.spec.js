import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('Button', () => {
	test('displays `Foo` as label', () => {
		const tree = renderer
			.create(<Button htmlId="defaultBtn" label="Foo" />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
