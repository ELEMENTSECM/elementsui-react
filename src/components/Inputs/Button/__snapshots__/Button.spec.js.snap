// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button displays \`Foo\` as label 1`] = `
<button
  aria-describedby={null}
  aria-label={undefined}
  aria-labelledby={null}
  aria-pressed={undefined}
  className="ms-Button ms-Button--primary root-36"
  data-is-focusable={true}
  disabled={undefined}
  onClick={undefined}
  type="button"
>
  <div
    className="ms-Button-flexContainer flexContainer-37"
  >
    <div
      className="ms-Button-textContainer textContainer-38"
    >
      <div
        className="ms-Button-label label-40"
        id="id__0"
      >
        Foo
      </div>
    </div>
  </div>
</button>
`;
